Lib/logging/handlers.py

```diff
@@ -196,9 +196,12 @@ def shouldRollover(self, record):
         if self.stream is None:                 # delay was set...
             self.stream = self._open()
         if self.maxBytes > 0:                   # are we rolling over?
+            pos = self.stream.tell()
```

Comment:
This section will clash with the other PR for gh-116267, perhaps wait until that is finalised?

Reply:
This PR is simpler, and it adds some tests for `delay=True`, so I think that it would be better to merge it first.